application/apps/rustcore/rs-bindings/src/js/session.rs

```diff
@@ -511,18 +540,23 @@ impl RustSession {
                 ));
             }
         };
-        self.op_channel
-            .0
-            .send(Operation::Assign {
+        let op_channel_tx = self.op_channel.0.clone();
+        async move {
+            match op_channel_tx.send(Operation::Assign {
```

Comment:
this does not look correct...what you are trying to do is to return a future. For that you should not artificially create an async block but rather use a combinator like `ready`
https://docs.rs/futures/0.3.15/futures/future/fn.ready.html

Reply:
Did you check what happens if you block in an async function? Is the js thread then also blocked? if so we need to wrap the blocking call in a `tokio::task::spawn_blocking`

Reply:
you are absolutely right, that `async {}` block doesn't necessary here at all as soon as `#[node_bindgen]` macro cares about wrapping into the future.